src/Keboola/GoogleAnalyticsExtractor/GoogleAnalytics/Client.php

```diff
@@ -198,4 +199,11 @@ public function getApiCallsCount()
     {
         return $this->apiCallsCount;
     }
+
+    public static function getDelayFn($base = 5000)
+    {
+        return function ($retries) use ($base) {
+            return (int) ($base * pow(2, $retries - 1) + rand(0, 500));
```

Comment:
just curious why the `rand(0, 500)` here?

Reply:
well, it's not really necessary here I think. It's basically to avoid collisions, when a few requests happen at the same time, they won't backoff exactly at the same times. If the API has a limit for number of requests per second.
https://en.wikipedia.org/wiki/Exponential_backoff

Reply:
ah, ok,  thanks for the explanation!